tests/test_edgeql_json.py

```diff
@@ -1024,3 +1024,139 @@ class TestEdgeQLJSON(tb.QueryTestCase):
             await self.con.execute(r"""
                 SELECT <json>b'foo';
             """)
+
+    async def test_edgeql_json_view_01(self):
+        await self.assert_query_result(r'''
+            WITH MODULE test
+            SELECT _ := json_get(JSONTest.j_array, '0')
+            ORDER BY _;
+
+            WITH MODULE test
+            SELECT _ := json_get(JSONTest.j_array, '10')
+            ORDER BY _;
+
+            WITH MODULE test
+            SELECT _ := json_get(JSONTest.j_array, {'-1', '4'})
+            ORDER BY _;
+
+            WITH MODULE test
+            SELECT _ := json_get(
+                JSONTest.data,
+                # Each of the variadic "steps" is a set, so we should
+                # get a cross-product of all the possibilities. It so
+                # happens that the only 2 valid paths are:
+                # - '4', 'b', 'bar', '1' -> null
+                # - '4', 'b', 'bar', '2' -> {"bingo": "42!"}
+                {'2', '4', '4'}, {'0', 'b'}, {'bar', 'foo'}, {'1', '2'}
+            )
+            ORDER BY _;
+        ''', [
+            [1, 2],
+            [],
+            [None, None, 1],
+            [None, None, {'bingo': '42!'}, {'bingo': '42!'}],
+        ])
+
+    async def test_edgeql_json_view_02(self):
+        await self.assert_query_result(r'''
+            WITH MODULE test
+            SELECT _ := json_get(
+                JSONTest.data,
+                # Each of the variadic "steps" is a set, so we should
+                # get a cross-product of all the possibilities. It so
+                # happens that the only 2 valid paths are:
+                # - '4', 'b', 'bar', '1' -> null
+                # - '4', 'b', 'bar', '2' -> {"bingo": "42!"}
+                {'2', '4', '4'}, {'0', 'b'}, {'bar', 'foo'}, {'1', '2'},
+                default := <json>'N/A'
+            )
+            ORDER BY _;
+        ''', [
+            # We expect 3 * 2 * 2 * 2 = 24 results per JSONTest.data.
+            # There are 3 non-empty data properties.
+            # Out of them all only 4 will have a valid path.
+            # So we expect 24 * 3 - 4 = 68 default 'N/A' results
+            [None, None] + ['N/A'] * 68 + [{'bingo': '42!'}, {'bingo': '42!'}],
```

Comment:
Worth mentioning that these `None` values here are actually the interpretation of JSON `null` values, and not an (invalid) leakage of SQL NULL.

Reply:
OK, I've updated comments in JSON tests and annotated occurrences of `None` that come from a legitimate `null`.